blockchain/blockchain-wallet-v4-frontend
# feat(add error boundaries to avoid fatal app crashes)

## Description
React recently changed the way errors are handled internally which means that a serious error during a components lifecycle will result in a blank white screen for users.  This work adds a new error boundary component which is able to wrap our component trees via wallet scenes and the public layout with error handling for a better UX.  I tried to put the component higher up in the component tree in hopes of only having to add it to a few spots in the app but i was unable to do it cleanly.  If you care for more details, I can explain outside of PR. 

We still need some design on the modals but this is a good start.

Also edited the modal provider component to allow for modals to not close when escape key is pressed or users clickoutside the modal.

Screenshots of the error modal are below.
<img width="1301" alt="screen shot 2018-06-14 at 9 25 39 am" src="https://user-images.githubusercontent.com/6364918/41423642-dbc3499e-6fc9-11e8-978e-efd966317f60.png">
<img width="1414" alt="screen shot 2018-06-14 at 11 42 09 am" src="https://user-images.githubusercontent.com/6364918/41423643-dbd1ddd8-6fc9-11e8-8994-07b91433a7f5.png">


## Change Type
Feature

## Testing Steps
Add some errors via `throw new Error()` in random components and see the component catch them.

## Code Checklist
